test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -2193,6 +2193,174 @@ Describe "Invoke-WebRequest tests" -Tags "Feature", "RequireAdminOnWindows" {
             $pathologicalRatio | Should -BeGreaterThan 5
         }
     }
+
+    Context 'WebSession persistence tests' {
+        $connWarning = 'WebSession properties changed: new Http connection will be required'
+
+        It 'Connection persistence maintained' {
+            $uri = Get-WebListenerUrl
+            $null = Invoke-WebRequest -Uri $uri -SessionVariable Session -WarningVariable warnings
+            $warnings | Should -BeNullOrEmpty
+
+            $null = Invoke-WebRequest -Uri $uri -WebSession $Session -WarningVariable warnings
+            $warnings | Should -BeNullOrEmpty
+
+            $null = Invoke-WebRequest -Uri $uri -WebSession $Session -WarningVariable warnings
+            $warnings | Should -BeNullOrEmpty
+
+            $null = Invoke-WebRequest -Uri $uri -WebSession $Session -WarningVariable warnings
+            $warnings | Should -BeNullOrEmpty
```

Comment:
Why duplicate?

Reply:
Mainly to prove it stays persistent for multiple requests. Might be better written as a foreach loop.